### PR TITLE
feat: soft delete endpoint for feature flags

### PR DESCRIPTION
**Pull request by xsadia**
This PR adds an endpoint to soft delete feature flags by adding the `deleted_at` property to a specific record

### CODE REVIEW

File: pkg/api/handlers/tests/feature_flag_handler_test.go [hunk 1]
```diff
@@ -673,10 +675,134 @@ func (suite *FeatureFlagHandlerTestSuite) TestRollbackUnauthorizedMissingPermiss
...
+			rule,
+		},
+	}
+	featureFlagRecord := models.NewFeatureFlagRecord(
+		featureFlagRequest.Name,
+		featureFlagRequest.DefaultValue,
+		featureFlagRequest.Type,
+		featureFlagRequest.Rules,
+		organization.ID,
+		user.ID,
```
Comment (LuisOw) [inline pkg/api/handlers/tests/feature_flag_handler_test.go:779]: Use the fixture

File: pkg/api/handlers/tests/feature_flag_handler_test.go [hunk 2]
```diff
@@ -673,10 +675,134 @@ func (suite *FeatureFlagHandlerTestSuite) TestRollbackUnauthorizedMissingPermiss
...
+	)
```
Comment (LuisOw) [inline pkg/api/handlers/tests/feature_flag_handler_test.go:716]: Use the fixture

File: pkg/api/handlers/feature_flag_handler.go [hunk 3]
```diff
@@ -272,6 +273,7 @@ func (ffh *FeatureFlagHandler) PatchFeatureFlag(c echo.Context) error {
 }
 
 func (ffh *FeatureFlagHandler) ApproveRevision(c echo.Context) error {
+	ffh.logger.Debug("aaaaaaaaaaaaaaaaaaaaaaa")
```
Comment (LuisOw) [inline pkg/api/handlers/feature_flag_handler.go:276]: Classic

Comment (xsadia) [reply pkg/api/handlers/feature_flag_handler.go:276]: oops

**Review by LuisOw — APPROVED** [review]

**Pull request merged**